dev/com.ibm.ws.jdbc/src/com/ibm/ws/rsadapter/impl/OracleHelper.java

```diff
@@ -885,7 +884,7 @@ public ConnectionResults getPooledConnection(final CommonDataSource ds, String u
 
     @FFDCIgnore(Exception.class)
     private void checkIBMJava8() throws ResourceException {
-        if (JavaInfo.majorVersion() == 8 && JavaInfo.vendor() == Vendor.IBM) {
+        if (JavaInfo.isAvailable("com.ibm.security.auth.module.Krb5LoginModule")) {
             // The Oracle JDBC driver prior to 21c does not support kerberos authentication on IBM JDK 8 because
             // it has dependencies to the internal Sun security APIs which don't exist in IBM JDK 8
             boolean ibmJdkSupported;
```

Comment:
The name of this method and variable should be changed to reflect what they actually check.

Further down in this method, the message produced is `KERBEROS_ORACLE_IBMJDK_NOT_SUPPORTED`. We might need to check that that message is still accurate.

Reply:
Perhaps a better name would be `assertIBMKerberosSupported`. I nearly did re-write this method because a) it should probably use `JavaInfo.isAvailable("oracle.jdbc.OracleDatabaseMetaData")` before trying to load it and b) catching Exception seems a little too broad.

I think these changes should be done as a follow up PR.

Reply:
I suggest to create a static method on JavaInfo... JavaInfo.isIBMJava, which just tries to load the class `com.ibm.security.auth.module.Krb5LoginModule`. Is it error prone to pass in the class names? Besides, in the future, maybe this class might be updated in IBM Java and then a lot of refactors will have to be performed.

Reply:
I did think about that too ... you don't want to call it `isIBMJava` because that isn't really what you're checking for. Something more like `isIBMKerberosSupported` would be better. There are so many areas that we could improve here. I think that this PR should be left as just the minimum required for 21.0.0.9.

Reply:
https://github.com/OpenLiberty/open-liberty/issues/18051

Reply:
See related conversation with Brent inhttps://github.com/OpenLiberty/open-liberty/issues/16366